### PR TITLE
Post Creation

### PR DESCRIPTION
**Pull request by shiara02**
### OVERVIEW

-Post creation (adding/storing and displaying all posts)

### NOTES

-displaying of images is not yet included

### SNIPPETS

image
image

### CODE REVIEW

File: src/app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+
+class PostController extends Controller
+{
+    public function index()
```
Comment (bpocrafael) [inline src/app/Http/Controllers/PostController.php]: why needed this?? nit remove

File: src/app/Http/Controllers/PostController.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        
+    }
+
+    public function create(Request $request)
```
Comment (bpocrafael) [inline src/app/Http/Controllers/PostController.php]: create a form request for this one

File: src/app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -14,7 +15,8 @@ public function show(): View
 
     public function home(): View
     {
-        return view('user/home');
+        $posts = Post::all();
```
Comment (bpocrafael) [inline src/app/Http/Controllers/UserController.php]: can you add a pagination for this one.

File: src/app/Models/Post.php [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Post extends Model
+{
+    use HasFactory;
+
+    protected $fillable = ['content', 'user_id' , 'image'];
```
Comment (bpocrafael) [inline src/app/Models/Post.php:12]: hmm can you add a relationship to the user table

File: src/resources/views/user/home.blade.php [hunk 5]
```diff
@@ -113,36 +86,18 @@
                     <h2 class="my-2 text-white text-center">Microblog Newsfeed</h2>
                 </div>
                 <ul class="list-group">
+                    @foreach ($posts as $post)
                     <li class="list-group-item" style="background-color: #FFFFF0">
                         <h5 class="card-title">User Name</h5>
-                        <p class="card-text">Sample1 post of the User</p>
```
Comment (bpocrafael) [inline src/resources/views/user/home.blade.php:118]: you can a create a component of a post

File: src/routes/web.php [hunk 6]
```diff
@@ -12,3 +13,4 @@
 Route::get('/profile/show', [UserController::class, 'show'])->name('profile.show');
 Route::post('/logout', [UserController::class, 'logout'])->name('user.logout');
 Route::get('/home', [UserController::class, 'home'])->name('user.home');
+Route::post('/create-post', [PostController::class, 'create']);
```
Comment (bpocrafael) [inline src/routes/web.php]: please fix this as per the naming standard format of creating a routes

**Review by bpocrafael — CHANGES_REQUESTED** [review]

File: src/database/migrations/2023_10_06_023551_create_posts_table.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+     */
+    public function up(): void
+    {
+        Schema::create('posts', function (Blueprint $table) {
+            $table->id();
+            $table->timestamps();
+            $table->string('content');
+            $table->string('image')->nullable();
+            $table->foreignId('user_id')->constrained();
+        });
```
Comment (hipe01mark) [inline src/database/migrations/2023_10_06_023551_create_posts_table.php]: create a foreign key with reference and an action on cascade:

File: src/routes/web.php (hunk 6 above)
Comment (hipe01mark) [reply src/routes/web.php]: +1 
https://webdevetc.com/blog/laravel-naming-conventions/

File: src/app/Http/Controllers/PostController.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+class PostController extends Controller
+{
+
+    public function create(PostRequest $request)
+    {
+        $post = new Post();
+        $post->content = $request->input('content');
+        $post->user_id = auth()->id();
+
+        if($request->hasFile('image')) {
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostController.php]: Follow PSR

File: src/app/Http/Controllers/PostController.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+            $destination_path = 'public/images';
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostController.php]: Use constant for static variables

File: src/app/Http/Controllers/UserController.php [hunk 10]
```diff
@@ -2,19 +2,24 @@
...
 use Illuminate\View\View;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Facades\Auth;
 
 class UserController extends Controller
 {
     public function show(): View
     {
-        return view('user/profile');
+        $user = Auth::user();
```
Comment (hipe01mark) [inline src/app/Http/Controllers/UserController.php]: Do you think you are using $user? If not, remove it.

File: src/app/Http/Requests/PostRequest.php [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
```
Comment (hipe01mark) [inline src/app/Http/Requests/PostRequest.php]: Remove docblock

File: src/resources/views/post/show.blade.php [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+        <h5 class="card-title">{{ $post->user->name }}</h5>
+        <p class="card-text">{{$post->content}}</p>
+        <img src="{{asset(''.$post->image)}}" alt="Post Image" class="img-fluid">
+        <div class="interaction mt-4">
+            <a href="#" class="btn btn-sm btn-primary like">Like</a>
+            <a href="#" class="btn btn-sm btn-success comment">Comment</a>
+            <a href="#" class="btn btn-sm btn-danger share">Share</a>
+        </div>
+    </li>
+    @endforeach
```
Comment (hipe01mark) [inline src/resources/views/post/show.blade.php]: Fix the indention. Follow PSR.

File: src/app/Http/Controllers/PostController.php (hunk 1 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Controllers/PostController.php (hunk 2 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Controllers/UserController.php (hunk 3 above)
Comment (shiara02) [reply src/app/Http/Controllers/UserController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Models/Post.php (hunk 4 above)
Comment (shiara02) [reply src/app/Models/Post.php:12]: I have resolved this comment, sir. Thank you! 🙇

File: src/resources/views/user/home.blade.php (hunk 5 above)
Comment (shiara02) [reply src/resources/views/user/home.blade.php:118]: I have resolved this comment, sir. Thank you! 🙇

File: src/routes/web.php (hunk 6 above)
Comment (shiara02) [reply src/routes/web.php]: I have resolved this comment, sirs. Thank you! 🙇

File: src/database/migrations/2023_10_06_023551_create_posts_table.php (hunk 7 above)
Comment (shiara02) [reply src/database/migrations/2023_10_06_023551_create_posts_table.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Controllers/PostController.php (hunk 8 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Controllers/PostController.php (hunk 9 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Controllers/UserController.php (hunk 10 above)
Comment (shiara02) [reply src/app/Http/Controllers/UserController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Requests/PostRequest.php (hunk 11 above)
Comment (shiara02) [reply src/app/Http/Requests/PostRequest.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/resources/views/post/show.blade.php (hunk 12 above)
Comment (shiara02) [reply src/resources/views/post/show.blade.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/resources/views/post/show.blade.php [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<ul class="list-group">
+    @foreach ($posts as $post)
+        <li class="list-group-item" style="background-color: #FFFFF0">
+            <h5 class="card-title">{{ $post->user->name }}</h5>
+            <p class="card-text">{{ $post->content }}</p>
+            <img src="{{ asset(''.$post->image) }}" alt="Post Image" class="img-fluid">
```
Comment (rafael-parayno) [inline src/resources/views/post/show.blade.php:6]: ''?

Comment (rafael-parayno) [reply src/resources/views/post/show.blade.php:6]: What if the post doesnt have an image?

Comment (shiara02) [reply src/resources/views/post/show.blade.php:6]: Regarding on the image sir, both storing and displaying po, i'm currently working on that po. May problem po kasi sa storage link that's why 'di ko pa po muna pinush through this PR.

**Review by rafael-parayno — APPROVED** [review]

**Review by hipe01mark — DISMISSED** [review]
LGTM 🙇  Just resolve other concers

File: src/app/Http/Controllers/PostController.php [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+namespace App\Http\Controllers;
+
+use App\Http\Requests\PostRequest;
+use App\Models\Post;
+
+class PostController extends Controller
+{
+    const DESTINATION_PATH = 'public/images';
+
+    public function create(PostRequest $request)
```
Comment (hipe-francis) [inline src/app/Http/Controllers/PostController.php]: shouldn't this method named store?

**Review by hipe-francis — CHANGES_REQUESTED** [review]

File: src/app/Http/Controllers/PostController.php [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+            $image = $request->file('image');
+            $image_name = $image->getClientOriginalName();
+            $path = $image->storeAs(self::DESTINATION_PATH, $image_name);
+
+            $input['image'] = $image_name;
+            $post->image = $path;
+        }
+
+        $post->save();
+        $returnRoute = $request->input('return_route', 'user.home');
```
Comment (bpocrafael) [inline src/app/Http/Controllers/PostController.php]: what is this?

File: src/app/Http/Requests/PostRequest.php [hunk 16]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class PostRequest extends FormRequest
+{
+    public function authorize(): bool
+    {
```
Comment (bpocrafael) [inline src/app/Http/Requests/PostRequest.php]: do you think this authorize function is doing anything in this context?

File: src/routes/web.php [hunk 17]
```diff
@@ -12,3 +13,4 @@
 Route::get('/profile/show', [UserController::class, 'show'])->name('profile.show');
 Route::post('/logout', [UserController::class, 'logout'])->name('user.logout');
 Route::get('/home', [UserController::class, 'home'])->name('user.home');
+Route::post('/post', [PostController::class, 'store'])->name('post.store');
```
Comment (bpocrafael) [inline src/routes/web.php]:
```suggestion
Route::post('/posts', [PostController::class, 'store'])->name('post.store');
```

File: src/app/Http/Controllers/PostController.php (hunk 15 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: after creating post po sir mag r'redirect po sa same page.

**Review by hipe-francis — DISMISSED** [review]
GEFM, you can merge once you resolve the remaining reviews.

File: src/app/Http/Controllers/PostController.php (hunk 15 above)
Comment (bpocrafael) [reply src/app/Http/Controllers/PostController.php]: why it is in the $request->input. just wondering

File: src/app/Http/Requests/PostRequest.php (hunk 16 above)
Comment (shiara02) [reply src/app/Http/Requests/PostRequest.php]: As of now po sir, wala pa po.

File: src/app/Http/Controllers/PostController.php (hunk 15 above)
Comment (hipe01mark) [reply src/app/Http/Controllers/PostController.php]: You can use redirect back withInput

https://laravel.com/docs/10.x/redirects

File: src/app/Http/Requests/PostRequest.php (hunk 16 above)
Comment (bpocrafael) [reply src/app/Http/Requests/PostRequest.php]: if none, remove it thanks

File: src/app/Http/Controllers/PostController.php (hunk 15 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: based po sa nabasa ko sir inaattempt po niyan na iretrieve yung value na associated sa return_route galing po sa request's input data. if nag-eexist daw po yung data, yun po yung gagamitin na value.

File: src/app/Http/Requests/PostRequest.php (hunk 16 above)
Comment (shiara02) [reply src/app/Http/Requests/PostRequest.php]: noted on that po, sir. thank you po.

**Review by hipe01mark — DISMISSED** [review]
LGTM 🙇

File: src/app/Http/Requests/PostRequest.php (hunk 16 above)
Comment (bpocrafael) [reply src/app/Http/Requests/PostRequest.php]: remove?

File: src/app/Http/Controllers/PostController.php (hunk 14 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Controllers/PostController.php (hunk 15 above)
Comment (shiara02) [reply src/app/Http/Controllers/PostController.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/app/Http/Requests/PostRequest.php (hunk 16 above)
Comment (shiara02) [reply src/app/Http/Requests/PostRequest.php]: I have resolved this comment, sir. Thank you! 🙇

File: src/routes/web.php (hunk 17 above)
Comment (shiara02) [reply src/routes/web.php]: I have resolved this suggestion, sir. Thank you! 🙇

**Review by bpocrafael — APPROVED** [review]

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**